tests/Doctrine/Tests/ORM/Cache/DefaultEntityHydratorTest.php

```diff
@@ -147,9 +147,39 @@ public function testBuildCacheEntryNonInitializedAssocProxy()
         $this->assertArrayHasKey('name', $cache->data);
         $this->assertArrayHasKey('country', $cache->data);
         $this->assertEquals(array(
-            'id'        => 11,
+            'id'        => 12,
             'name'      => 'Bar',
             'country'   => new AssociationCacheEntry(Country::CLASSNAME, array('id' => 11)),
         ), $cache->data);
     }
-}
\ No newline at end of file
+
+	public function testCacheEntryWithWrongIdentifierType()
+	{
+		$proxy          = $this->em->getReference(Country::CLASSNAME, 11);
+		$entity         = new State('Bat', $proxy);
+		$uow            = $this->em->getUnitOfWork();
+		$entityData     = array('id'=> 12, 'name'=>'Bar', 'country' => $proxy);
+		$metadata       = $this->em->getClassMetadata(State::CLASSNAME);
+		$key            = new EntityCacheKey($metadata->name, array('id'=>'12'));
```

Comment:
Yes, as described in the issue on Jira, if you do `$repository->find('1')` the cache entry key instance contains the string value while entity contains integer value fetched from database.